xxh_aarch64dispatch.S

```diff
@@ -0,0 +1,1043 @@
+#include "asmdefs.h"
+
+.data
+
+xxh3_prime_1:	.dword	0x9E3779B1U
+
+.text
+
+.macro init_index
+	mov	z1.d, #2
+	pfalse	p1.b
+	index	z7.d, #1, #1
+	trn1	p1.d, p1.d, p7.d
+	sub	z7.d, p1/m, z7.d, z1.d
+.endm
+
+.macro calc_acc512 acc
+	// swapped = SWAP(input)
+	tbl	z5.d, {z1.d}, z7.d
+	// mixed = input EOR secret
+	eor	z1.d, p7/m, z1.d, z2.d
+	mov	z6.d, z1.d
+	// mixed_lo = mixed AND 0xffffffff
+	and	z1.d, p7/m, z1.d, z3.d
+	// mixed_hi = mixed >> 32
+	lsr	z6.d, p7/m, z6.d, z4.d
+	// mixed_lo = mixed_hi * acc + mixed_lo
+	mad	z1.d, p7/m, z6.d, \acc
```

Comment:
You should be able to avoid the temporary like this:
```asm
        mad z1.d, p7/m, z6.d, z5.d
        add \acc, p7/m, \acc, z1.d
```

Reply:
Thanks a lot. It could save one instruction.